### ISSUE TITLE
Operator parameters and stacked measurements coexist poorly in HDF5Dataset

### ISSUE THREAD

**Issue by jscanvic** [deepinv/deepinv#767]
### Environment

Latest

### Describe the bug

The HDF5Dataset class supports ground truths measured with multiple forward operators combined using `StackedPhysics`. In that case, it stores the different measurements in members of the form `y{physics_idx}_{split_name}` in the h5 file. As it turns out, this interacts poorly with another feature the class has, which is support for storing forward operator parameters (PSFs, masks) enabled by the parameter `load_physics_generator_params`.

Indeed, in the current implementation, any member of the form `{something}_{split_name}` where `{something}` is different from `x` and `y` is considered to be a parameter the forward operator has.

This means that, in the presence of stacked measurements with `load_physics_generator_params` enabled, the member `y0_train` is incorrectly interpreted as a forward operator parameter, in addition to being interpreted as a part of the stacked measurements.

**Fix**

One way to fix the bug is to assume that a h5 member that forms a valid stacked measurement cannot be a physics parameter and add the check in the code.

### Steps to reproduce

Adapted from https://github.com/deepinv/deepinv/blob/1a4803087be766b564441c50f0b0887b88d808aa/examples/physics/demo_mri_tour.py

```python
import deepinv as dinv
import torch, torchvision
from torch.utils.data import DataLoader

device = dinv.utils.get_freer_gpu() if torch.cuda.is_available() else "cpu"
rng = torch.Generator(device=device).manual_seed(0)

transform = torchvision.transforms.Resize(128)
knee_dataset = dinv.datasets.SimpleFastMRISliceDataset(
    dinv.utils.get_data_home(),
    anatomy="knee",
    transform=transform,
    train=True,
    download=True,
)
brain_dataset = dinv.datasets.SimpleFastMRISliceDataset(
    dinv.utils.get_data_home(),
    anatomy="brain",
    transform=transform,
    train=True,
    download=True,
)

img_size = knee_dataset[0].shape[-2:]  # (128, 128)

physics_generator = dinv.physics.generator.GaussianMaskGenerator(
    img_size=img_size, acceleration=4, rng=rng, device=device
)
mask = physics_generator.step()["mask"]

physics = dinv.physics.MRI(mask=mask, img_size=img_size, device=device)
physics = dinv.physics.StackedPhysics([physics, physics])

dataset_path = dinv.datasets.generate_dataset(
    train_dataset=knee_dataset,
    test_dataset=brain_dataset,
    val_dataset=None,
    physics=physics,
    physics_generator=physics_generator,
    save_physics_generator_params=True,
    overwrite_existing=True,
    device=device,
    save_dir=dinv.utils.get_data_home(),
    batch_size=1,
    supervised=False,
)

test_dataset = dinv.datasets.HDF5Dataset(
    dataset_path, split="test", load_physics_generator_params=True
)

# This first assert passes
assert "x_test" in test_dataset.hd5, "There are test ground truths in the dataset"

x, y, params = test_dataset[0]

print(params.keys()) # dict_keys(['mask', 'y0', 'y1'])
```

### Expected behavior

It should print `dict_keys(['mask'])`
